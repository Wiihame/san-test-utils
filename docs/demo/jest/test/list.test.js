/**
 * @file jest demo file
 **/
import {shallowMount} from '@baidu/san-test-utils';
import list from '@/components/list';

describe('list', () => {
    it('renders li for each item in items', () => {
        const items = ['1', '2'];
        const wrapper = shallowMount(list, {
            data: {items}
        });
        expect(wrapper.findAll('li')).toHaveLength(items.length);
    });

    it('matches snapshot', () => {
        const items = ['item 1', 'item 2'];
        const wrapper = shallowMount(list, {
            data: {items}
        });
        expect(wrapper.html()).toMatchSnapshot();
    });
});
