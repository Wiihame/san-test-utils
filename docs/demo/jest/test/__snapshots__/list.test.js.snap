// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list matches snapshot 1`] = `
"<ul>
        <li>
            item 1
        </li><li>
            item 2
        </li><!--4-->
    </ul>"
`;
